include/kodlab_mjbots_sdk/joint_moteus.h

```diff
@@ -103,129 +114,193 @@ class JointMoteus: public JointBase
          * @brief Construct a new Joint Moteus object with a MoteusJointConfig struct
          * 
          * @param config MoteusJointConfig input
          */
-        JointMoteus(MoteusJointConfig config)
-            : JointBase( config.name,
-                         config.direction,
-                         config.zero_offset,
-                         config.gear_ratio,
-                         config.max_torque,
-                         config.pos_min,
-                         config.pos_max,
-                         config.soft_start_duration_ms),
-              can_id_( config.can_id),
-              can_bus_( config.can_bus),
-              moteus_kp_(config.moteus_kp),
-              moteus_kd_(config.moteus_kd){}
+        JointMoteus(MoteusJointConfig config);
         
+        /**
+         * @brief Update the joint of the moteus. Converts rot/s to rad/s and saves other state values.
+         * 
+         * @param reply_message reply message sent by moteus to pi3hat
+         */
+        void UpdateMoteus(::mjbots::moteus::QueryResult reply_message);
+
         /**
          * @brief Destroy the JointMoteus object, virtual to ensure proper 
          * destruction of derived classes
          */
         virtual ~JointMoteus(){}
 
         /**
-         * @brief Update the joint of the moteus. Converts rot/s to rad/s and saves mode
+         * @brief Determines whether open loop options should be enabled
          * 
-         * @param reply_pos position reported by moteus [rot]
-         * @param reply_vel velocity reported by moteus [rot/s]
-         * @param reply_torque torque measured by the moteus [Nm]
-         * @param mode      moteus mode
+         * @return bool 
          */
-        void UpdateMoteus(float reply_pos, float reply_vel, float reply_torque, ::mjbots::moteus::Mode mode ){
-            UpdateState( 2 * M_PI * reply_pos, 2 * M_PI * reply_vel, reply_torque);
-            mode_ = mode;
-        }
+        bool is_open_loop();
 
         /**
          * @brief Get the can id 
          * 
          * @return int 
          */
-        int get_can_id() const {return can_id_;}
+        int get_can_id() const;
 
         /**
          * @brief Get the can bus object
          * 
          * @return int 
          */
-        int get_can_bus() const {return can_bus_;}
+        int get_can_bus() const;
+
+        /**
+         * @brief Get the QueryCommand object
+         * 
+         * @return const ::mjbots::moteus::QueryCommand
+         */
+        const ::mjbots::moteus::QueryCommand get_query_command() const;
 
         /**
          * @brief Get the mode reference 
          * 
          * @return const ::mjbots::moteus::Mode& 
          */
-        const ::mjbots::moteus::Mode & get_mode_reference()   const {return mode_; }
+        const ::mjbots::moteus::Mode & get_mode_reference() const;
+
+        /**
+         * @brief Get the mode 
+         * 
+         * @return const ::mjbots::moteus::Mode
```

Comment:
These ai generated comments are not useful. Either make them useful or remove them.